src/AppBundle/Controller/AdminStatsController.php

```diff
@@ -1,103 +1,264 @@
+                ]
+            );
+        }
+
+        // Initialize variables - prevents variable undefined errors
+        $adminIdArr = [];
+        $adminsWithoutAction = 0;
+        $adminsWithoutActionPct = 0;
 
+        // Pull the labs helper, API helper, and database.  Then check if we can
+        // use this tool
         $lh = $this->get("app.labs_helper");
+        $api = $this->get("app.api_helper");
+        $conn = $this->get('doctrine')->getManager("replicas")->getConnection();
 
         $lh->checkEnabled("adminstats");
 
+        // Load the database information for the tool
         $dbValues = $lh->databasePrepare($project, "AdminStats");
 
         $dbName = $dbValues["dbName"];
         $wikiName = $dbValues["wikiName"];
         $url = $dbValues["url"];
 
-        $users = [
-            "Matthew" => [
-                "group" => "A", "delete" => "0", "restore" => "0", "block" => "0", "unblock" => "0",
-                "protect" => "0", "unprotect" => "0", "import" => "0", "rights"=>"0", "total"=>"0"
-            ],
-            "Aexandra" => [
-                "group" => "ABC", "delete" => "0", "restore" => "0", "block" => "0",
-                "unblock" => "0", "protect" => "0", "unprotect" => "0", "import" => "0",
-                "rights"=>"0", "total"=>"0"
-            ]
-        ];
+        // Generate a diff for the dates - this is the number of days we're spanning.
+        $days = date_diff(new \DateTime($end), new \DateTime($start))->days;
+
+        // Pull the admins from the API, for merging.
+        $data = $api->getAdmins($project);
+
+        // Get admin ID's, used to account for inactive admins
+        $user_groups_table = $lh->getTable("user_groups", $dbName);
+        $ufg_table = $lh->getTable("user_former_groups");
+        $query = "
+    Select ug_user as user_id
+    FROM $user_groups_table
+    WHERE ug_group = 'sysop'
+    UNION
+    SELECT ufg_user as user_id
+    FROM $ufg_table
+    WHERE ufg_group = 'sysop'
+    ";
+
+        $res = $conn->prepare($query);
+        $res->execute();
+
+        // Iterate over query results, loading each user id into the array
+        while ($row = $res->fetch()) {
+            $adminIdArr[] = $row["user_id"] ;
+        }
+
+        // Set the query results to be useful in a sql statement.
+        $adminIds = implode(',', $adminIdArr);
 
-        dump($users);
+        // Load up the tables we need and run the mega query.
+        // This query provides all of the statistics
+        $userTable = $lh->getTable("user", $dbName);
+        $loggingTable = $lh->getTable("logging", $dbName);
 
-        return $this->render("adminStats/result.html.twig", [
-            'base_dir' => realpath($this->getParameter('kernel.root_dir').'/..'),
-            "xtPage" => "adminstats",
-            "xtPageTitle" => "tool_adminstats",
-            "xtSubtitle" => "tool_adminstats_desc",
+        // TODO: Fix this - inactive admins aren't getting shown
+        $query = "
```

Comment:
We do need the usernames since some users are not admins but have made admin actions (we get the current admins with `$api->getAdmins($project);`). Would removing the `ORDER BY` help? My SQL inexperience suggests that might involve a full table scan, when we could do the sorting with PHP.